hnn_core/tests/test_network.py

```diff
@@ -702,15 +715,24 @@ def test_network_connectivity():
         kwargs = kwargs_default.copy()
         kwargs[arg] = item
         indices = pick_connection(**kwargs)
+        if item is None:
+            assert len(indices) == 0
         for conn_idx in indices:
-            if (arg == 'src_gids' or arg == 'target_gids') and \
-                    isinstance(item, str):
-                assert np.all(np.in1d(net.connectivity[conn_idx][arg],
-                              net.gid_ranges[item]))
-            elif item is None:
-                pass
+            if isinstance(item, range):
+                # arg specifies a subset of item gids (within range)
+                net.connectivity[conn_idx][arg] <= set(item)
+            elif isinstance(item, str):
+                if arg in {'src_gids', 'target_gids'}:
+                    # arg specifies a subset of item gids (within gid_ranges)
+                    assert (net.connectivity[conn_idx][arg] <=
+                            set(net.gid_ranges[item]))
+                else:
+                    # arg and item specify equivalent string descriptors for
+                    # this connection type
+                    assert net.connectivity[conn_idx][arg] == item
             else:
-                assert np.any(np.in1d([item], net.connectivity[conn_idx][arg]))
+                # arg specifies a superset of item gids
+                assert set(net.connectivity[conn_idx][arg]) >= {item}
```

Comment:
```suggestion
                assert set(net.connectivity[conn_idx][arg]).issuperset({item})
```

Reply:
read this again and understand the logic now

Reply:
I updated all of these to explicitly use a set method instead of the inequality operator shorthand.